File: backend/internal/models/faq.go
package models

// Faq is one question on the public FAQ page, shown in ascending Order.
type Faq struct {
	// ID refers to document id and is not written upstream
	ID string `json:"id" firestore:"-"`

	Question string `json:"question" firestore:"question"`
	Answer   string `json:"answer"   firestore:"answer"`
	Order    int    `json:"order"    firestore:"order"`
}

File: app/src/api/faq.ts
// sat-app/app/src/api/faq.ts
import { FAQ } from "@/types/models";
import { apiFetch, readError } from "./client";

interface ListFaqsResponse {
  items: Array<{
      id: string;
      question: string;
      answer: string;
      order: number;

  }>;

}

/**
 * Fetches public FAQs from the Go backend: GET /api/v1/faq
 */
export async function listPublicFAQs(): Promise<FAQ[]> {
  const response = await apiFetch("/faqs", {}, "none");
  if (!response.ok) {
    const errorMsg = await readError(response, "Could not load FAQs");
    throw new Error(errorMsg);
  }

  const body = (await response.json()) as ListFaqsResponse;
  return (body.items || []).map((item, index) => ({
    id: item.id,
    question: item.question,
    answer: item.answer,
    order: item.order ?? index + 1,
  }));
}
